tests/generic_config_updater/field_operation_validator_test.py

```diff
@@ -14,6 +14,80 @@
+        patch_element = {"path": "/PORT/Ethernet3", "op": "remove"}
+        assert generic_config_updater.field_operation_validators.port_config_update_validator(patch_element) == True
+
+    @patch("generic_config_updater.field_operation_validators.read_statedb_entry", mock.Mock(return_value="rs, fc"))
+    def test_port_config_update_validator_invalid_fec_existing_state_db(self):
+        patch_element = {"path": "/PORT/Ethernet3/fec", "op": "add", "value": "asf"}
+        assert generic_config_updater.field_operation_validators.port_config_update_validator(patch_element) == False
+    
+    @patch("generic_config_updater.field_operation_validators.read_statedb_entry", mock.Mock(return_value="rs, fc"))
+    def test_port_config_update_validator_invalid_fec_existing_state_db_nested(self):
+        patch_element = {"path": "/PORT", "op": "add", "value": {"Ethernet3": {"alias": "Eth0", "fec": "none"}, "Ethernet4": {"alias": "Eth4", "fec": "fs"}}}
+        assert generic_config_updater.field_operation_validators.port_config_update_validator(patch_element) == False
+    
+    @patch("generic_config_updater.field_operation_validators.read_statedb_entry", mock.Mock(return_value="rs, fc"))
+    def test_port_config_update_validator_valid_fec_existing_state_db_nested(self):
+        patch_element = {"path": "/PORT", "op": "add", "value": {"Ethernet3": {"alias": "Eth0", "fec": "fc"}}}
+        assert generic_config_updater.field_operation_validators.port_config_update_validator(patch_element) == True
+    
+    @patch("generic_config_updater.field_operation_validators.read_statedb_entry", mock.Mock(return_value="rs, fc"))
+    def test_port_config_update_validator_valid_fec_existing_state_db_nested_2(self):
+        patch_element = {"path": "/PORT", "op": "add", "value": {"Ethernet3": {"alias": "Eth0", "fec": "rs"}, "Ethernet4": {"alias": "Eth4", "fec": "fc"}}}
+        assert generic_config_updater.field_operation_validators.port_config_update_validator(patch_element) == True
+    
+    @patch("generic_config_updater.field_operation_validators.read_statedb_entry", mock.Mock(return_value="rs, fc"))
+    def test_port_config_update_validator_valid_fec_existing_state_db(self):
+        patch_element = {"path": "/PORT/Ethernet3/fec", "op": "add", "value": "rs"}
+        assert generic_config_updater.field_operation_validators.port_config_update_validator(patch_element) == True
+
+    @patch("generic_config_updater.field_operation_validators.read_statedb_entry", mock.Mock(return_value=""))
+    def test_port_config_update_validator_valid_fec_no_state_db(self):
+        patch_element = {"path": "/PORT/Ethernet3", "op": "add", "value": {"fec": "rs"}}
+        assert generic_config_updater.field_operation_validators.port_config_update_validator(patch_element) == True
+    
+    @patch("generic_config_updater.field_operation_validators.read_statedb_entry", mock.Mock(return_value=""))
+    def test_port_config_update_validator_invalid_fec_no_state_db(self):
```

Comment:
>test_port_config_update_validator_invalid_fec_no_state_db [](http://example.com/codeflow?start=8&length=57)

Could you add unit test or sonic-mgmt test for config CLI with yang validation?

Reply:
Yes, I can add sonic-mgmt test to test config CLI with yang validation. 

Test for success scenario: 
1. Enable YANG ConfigDB validation feature (via CLI)
2. Use config CLI to configure speed/fec to valid value based on YANG validation and StateDB
3. Ensure that modification goes through 

Test for failure scenario: 
1. Enable YANG ConfigDB validation feature (via CLI)
2. Use config CLI to configure speed/fec to invalid value based on StateDB (but valid value based on YANG)
3. Ensure that output throws error "Modification of PORT table is illegal- validating function generic_config_updater.field_operation_validators.port_config_update_validator returned False"